workspaces/adventure-pack/goodies/java/src/traverse_post_order/AP.java

```diff
@@ -0,0 +1,63 @@
+package traverse_post_order;
+
+import iterable_stream.IterableStream;
+import java.util.NoSuchElementException;
+import java.util.Stack;
+import pair.Pair;
+import simple_iterator.SimpleIterator;
+
+public final class AP {
+
+  /**
+   * Get {@link IterableStream iterable stream} of all the {@link TreeNode nodes} in the binary tree
+   * in post-order.
+   * @param root Binary tree root {@link TreeNode node}.
+   * @return The {@link IterableStream iterable stream} in post-order.
+   */
+  public static IterableStream<TreeNode> traversePostOrder(
+    final TreeNode root
+  ) {
+    final Stack<Pair<TreeNode, Boolean>> stack = new Stack<
+      Pair<TreeNode, Boolean>
+    >();
+    stack.push(new Pair<>(root, false));
+    return IterableStream.from(
+      SimpleIterator.toIterator(() -> {
+        while (!stack.isEmpty()) {
+          Pair<TreeNode, Boolean> peek = stack.pop();
```

Comment:
I have mixed feelings about the variable name `peek`, since we didn't just peek, we popped 🙂 

In the other traversal implementation, I went for a verbose but hopefully clear name (`nodeAndDidTraverseLeftChild`). So I'd advocate for a similar name here (like `nodeAndDidTraverseChildren`). If you have a suggestion for another name I would be interested in, but let's then make sure we also change the other Java traversal implementation, so that we stay consistent.

Reply:
Yeah, agreed. Resolved.